### PR TITLE
frame start

### PR DESCRIPTION
**Pull request by riomus**

### CODE REVIEW

File: embeddings/data/Dataset.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from typing import TypeVar, Generic
+from abc import ABC
+
+Data = TypeVar("Data")
+
+
+class DataSet(ABC, Generic[Data]):
```
Comment (laugustyniak) [inline embeddings/data/Dataset.py:7]: maybe just a `Dataset`? wdyt?

File: embeddings/transformation/Transformation.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import abc
+from typing import TypeVar, Generic
+from abc import ABC
+
+Input = TypeVar("Input")
+Output = TypeVar("Output")
+
+
+class Transformation(ABC, Generic[Input, Output]):
```
Comment (laugustyniak) [inline embeddings/transformation/Transformation.py:9]: do we want to add explicitly something like `StaticEmbeddings(Transformation)`?

File: embeddings/data/Dataset.py (hunk 1 above)
Comment (albert097) [inline embeddings/data/Dataset.py:7]: typically class like this is named 'Dataset'

File: embeddings/transformation/Transformation.py (hunk 2 above)
Comment (laugustyniak) [reply embeddings/transformation/Transformation.py:9]: or rather extract separate step for Embeddings

**Review by laugustyniak — CHANGES_REQUESTED** [review]

File: embeddings/data/Dataset.py (hunk 1 above)
Comment (laugustyniak) [reply embeddings/data/Dataset.py:7]: hahah I got the same :D

File: embeddings/transformation/Transformation.py (hunk 2 above)
Comment (riomus) [reply embeddings/transformation/Transformation.py:9]: separated model and task

File: embeddings/data/Dataset.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+from typing import TypeVar, Generic
+from abc import ABC
+
+Data = TypeVar("Data")
+
+
+class Dataset(ABC, Generic[Data]):
```
Comment (laugustyniak) [inline embeddings/data/Dataset.py:7]: outstanding name :D

File: embeddings/pipeline/PipelineBuilder.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+CreationEvaluationResult = TypeVar("CreationEvaluationResult")
+
+
+class PipelineBuilder(
+    ABC,
+    Generic[Data, LoaderResult, TransformationResult, ModelResult, TaskResult, EvaluationResult],
+):
+    def __init__(
+        self,
+        data_set: Optional[Dataset[Data]] = None,
```
Comment (laugustyniak) [inline embeddings/pipeline/PipelineBuilder.py:34]: maybe `dataset` everywhere where is it data-related?

**Review by laugustyniak — CHANGES_REQUESTED** [review]

**Review by ktagowski — APPROVED** [review]

**Review by albert097 — APPROVED** [review]

File: embeddings/embedding/Embedding.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import abc
+from typing import TypeVar, Generic
+from abc import ABC
+
+Input = TypeVar("Input")
+Output = TypeVar("Output")
+
+
+class Embedding(ABC, Generic[Input, Output]):
```
Comment (laugustyniak) [inline embeddings/embedding/Embedding.py:9]: :)

**Review by laugustyniak — APPROVED** [review]

**Pull request merged**